src/services/state-persistor/utils/FeatureStyleHelper.ts

```diff
@@ -0,0 +1,911 @@
+import Feature from 'ol/Feature.js'
+import Map from 'ol/Map.js'
+import { Coordinate } from 'ol/coordinate.js'
+import ngeoFormatFeatureProperties from './FeatureProperties'
+import ngeoGeometryType from './GeometryTypes'
+import * as olArray from 'ol/array.js'
+import * as olColor from 'ol/color.js'
+import * as olExtent from 'ol/extent.js'
+import olGeomLineString from 'ol/geom/LineString.js'
+import olGeomMultiLineString from 'ol/geom/MultiLineString.js'
+import olGeomMultiPoint from 'ol/geom/MultiPoint.js'
+import olGeomPoint from 'ol/geom/Point.js'
+import olGeomPolygon from 'ol/geom/Polygon.js'
+import olGeomMultiPolygon from 'ol/geom/MultiPolygon.js'
+import olStyleCircle from 'ol/style/Circle.js'
+import olStyleFill from 'ol/style/Fill.js'
+import olStyleRegularShape from 'ol/style/RegularShape.js'
+import olStyleStroke from 'ol/style/Stroke.js'
+import olStyleStyle from 'ol/style/Style.js'
+import olStyleText from 'ol/style/Text.js'
+import * as olProj from 'ol/proj.js'
+import {
+  getFormattedLength,
+  getFormattedArea,
+  getFormattedAzimutRadius,
+  getFormattedPoint,
+} from '@/services/common/measurement.utils'
+
+class FeatureStyleHelper {
+  private precision_: number | undefined = undefined
+  private projection_: olProj.Projection | undefined
+  private decimals_ = 3
+  private pointFilterFn_: Function | null = null
+  private pointFilterArgs_: number[][] = []
+
+  /**
+   * @param {!ol.proj.Projection} projection Projection.
+   * @export
+   */
+  public setProjection(projection: olProj.Projection) {
+    this.projection_ = projection
+  }
+
+  // === STYLE METHODS ===
+
+  /**
+   * Set the style of a feature using its inner properties and depending on
+   * its geometry type.
+   * @param {!ol.Feature} feature Feature.
+   * @param {boolean=} opt_select Whether the feature should be rendered as
+   *     selected, which includes additional vertex and halo styles.
+   * @export
+   */
+  public setStyle(feature: Feature, opt_select: boolean) {
+    const styles = this.getStyle(feature)
+    if (opt_select) {
+      if (this.supportsVertex_(feature)) {
+        styles.push(this.getVertexStyle())
+      }
+      styles.unshift(this.getHaloStyle_(feature))
+    }
+    // TODO TS hack as any
+    feature.setStyle(styles as any)
+  }
+
+  /**
+   * Create and return a style object from a given feature using its inner
+   * properties and depending on its geometry type.
+   * @param {!ol.Feature} feature Feature.
+   * @return {!Array.<!ol.style.Style>} The style object.
+   * @export
+   */
+  public getStyle(feature: Feature) {
+    const type = this.getType(feature)
+    let style
+
+    switch (type) {
+      case ngeoGeometryType.LINE_STRING:
+        style = this.getLineStringStyle_(feature)
+        break
+      case ngeoGeometryType.POINT:
```

Comment:
why keeping reference to ngeo? maybe if you really want to keep a ref to ngeo you should move all this functionality outside the src folder of lux v4 and create a deidicated module.

Reply:
removed